### PR TITLE
Update to run zap test on legal application

### PR DESCRIPTION
**Pull request by robertparkinson**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-2413

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -66,6 +67,45 @@ node {
         sh "${dockerCompose} down --remove-orphans"
       }
     }
+
+    stage('Checkout Legal') {
+      deleteDir()
```
Comment (timja) [inline Jenkinsfile]: shouldn't need to do this

File: Jenkinsfile [hunk 2]
```diff
@@ -66,6 +67,45 @@ node {
...
+      checkout scm
```
Comment (timja) [inline Jenkinsfile]: shouldn't need to do this

File: bin/set-legal-scanning-exclusions.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    exit 1
+  else
+    RETRIES=$(($RETRIES+1))
+  fi
+done
+
+# Exclude external domains from scanning
+echo "Setting scanning exclusions"
+
+docker-compose ${OPTIONS} exec $@ zap-proxy zap-cli exclude '.*www.payments.service.gov.uk.*'
```
Comment (timja) [inline bin/set-legal-scanning-exclusions.sh:32]: legal app doesn't use pay

File: docker-compose-legal.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+version: '2.1'
+
+services:
+  zap-proxy:
```
Comment (timja) [inline docker-compose-legal.yml]: this looks like it should be in a shared docker-compose file, I can't see any changes here.

**Review by timja — CHANGES_REQUESTED** [review]

File: docker-compose-legal.yml (hunk 4 above)
Comment (rlewan) [reply docker-compose-legal.yml]: We probably need shared compose file for T1 and T2 in general, but that's a bigger task. Maybe we could leave this as is and get rid of all such duplications at once?

Comment (timja) [reply docker-compose-legal.yml]: I think for the work required not to duplicate this we should do this now

Comment (rlewan) [reply docker-compose-legal.yml]: Fine with me 👌

File: bin/set-legal-scanning-exclusions.sh (hunk 3 above)
Comment (robertparkinson) [reply bin/set-legal-scanning-exclusions.sh:32]: we're going to be including it soon so no harm leaving it in there.

File: Jenkinsfile [hunk 5]
```diff
@@ -57,13 +57,40 @@ node {
...
+        stage('Update images') {
+          sh "${legalDockerCompose} pull"
+        }
+
+        stage('Start & setup legal environment') {
+          sh "${legalDockerCompose} up -d zap-proxy remote-webdriver legal-frontend"
+          sh "./bin/set-legal-scanning-exclusions.sh ${execParams}"
+        }
+
+        stage('Run user journey through ZAP') {
```
Comment (rlewan) [inline Jenkinsfile]: Can you please rename this and corresponding citizen frontend stages for clarity?

File: Jenkinsfile [hunk 6]
```diff
@@ -57,13 +57,42 @@ node {
...
+        stage('Checkout Legal Integration Tests') {
+          checkoutLegalIntegrationTests()
+        }
+
+        stage('Update images') {
+          sh "${legalDockerCompose} pull"
+        }
+
+        stage('Start & setup legal environment') {
+          sh "docker-compose stop zap-proxy"
```
Comment (timja) [inline Jenkinsfile]: I think this should be: `${dockerCompose}`

File: Jenkinsfile [hunk 7]
```diff
@@ -57,13 +57,42 @@ node {
...
+          sh "docker-compose rm -f zap-proxy"
```
Comment (timja) [inline Jenkinsfile]: same here

**Review by timja — APPROVED** [review]

**Pull request merged**